code/iaas/healthcheck/src/main/java/io/cattle/iaas/healthcheck/service/impl/HealthcheckServiceImpl.java

```diff
@@ -259,17 +260,33 @@ protected void createHealthCheckHostMaps(HealthcheckInstanceType instanceType, l
                 HealthcheckInstance.class, healthInstance.getId());
         List<? extends Host> availableActiveHosts = allocatorDao.getActiveHosts(healthInstance.getAccountId());
         
+        // skip non-linux hosts
+        Iterator<? extends Host> it = availableActiveHosts.iterator();
+        while (it.hasNext()) {
+            Host host = it.next();
+            Map<String, Object> data = (Map<String, Object>) host.getData();
+            if (data.containsKey("fields")) {
+                Map<String, Object> fields = (Map<String, Object>) data.get("fields");
+                if (fields.containsKey("labels")) {
+                    Map<String, String> labels =(Map<String, String>) fields.get("labels");
+                    if (labels.containsKey("io.rancher.host.os") && !labels.get("io.rancher.host.os").equals("linux")) {
```

Comment:
Here are the associated changes:
https://github.com/LLParse/agent/blob/windows/core/hostInfo/os_c.go#L56-L58
https://github.com/LLParse/rancher-catalog/blob/windows/infra-templates/windows-healthcheck/0/docker-compose.yml#L7